howto/pubsub-namespaces/README.md

```diff
@@ -0,0 +1,107 @@
+# Using PubSub across multiple namespaces
+
+In some scenarios, applications can be spread across namespaces and share a queue or topic via PubSub. In this case, the PubSub component must be provisioned on each namespace.
+
+In this example, we will use the [PubSub sample](https://github.com/dapr/samples/tree/master/4.pub-sub). Redis installation and the subscribers will be in `namespace-a` while the publisher UI will be on `namespace-b`. This solution should also work if Redis was installed on another namespace or if we used a managed cloud service like Azure ServiceBus.
+
+The table below shows which resources are deployed to which namespaces:
+| Resource | namespace-a | namespace-b |
+|-|-|-|
+| Redis master | X ||
+| Redis slave | X ||
+| Dapr's PubSub component | X | X |
+| Node subscriber | X ||
+| Python subscriber | X ||
+| React UI publisher | | X|
+
+## Pre-requisites
+
+* [Dapr installed](https://github.com/dapr/docs/blob/master/getting-started/environment-setup.md.) on any namespace since Dapr works at the cluster level.
+
+## Setup `namespace-a`
+
+Create namespace and switch kubectl to use it.
+```
+kubectl create namespace namespace-a
+kubectl config set-context --current --namespace=namespace-a
+```
+
+Install Redis (master and slave) on `namespace-a`:
+```
+helm repo add bitnami https://charts.bitnami.com/bitnami
+helm repo update
+helm install redis bitnami/redis
+export REDIS_PASSWORD=$(kubectl get secret --namespace namespace-a redis -o jsonpath="{.data.redis-password}" | base64 --decode)
+```
+
+Now, configure `deploy/redis.yaml`, paying attention to the hostname containing `namespace-a`.
+
+```
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: messagebus
+spec:
+  type: pubsub.redis
+  metadata:
+  - name: "redisHost"
+    value: "redis-master.namespace-a.svc:6379"
+  - name: "redisPassword"
+    value: "YOUR_PASSWORD"
+```
+
+Deploy resources to `namespace-a`:
+```
+kubectl apply -f deploy/redis.yaml
+kubectl apply -f deploy/node-subscriber.yaml
+kubectl apply -f deploy/python-subscriber.yaml
+```
+
+## Setup `namespace-b`
+
+Create namespace and switch kubectl to use it.
+```
+kubectl create namespace namespace-b
+kubectl config set-context --current --namespace=namespace-b
+```
+
+Deploy resources to `namespace-b`, including the Redis component:
+```
+kubectl apply -f deploy/redis.yaml
+kubectl apply -f deploy/react-form.yaml
+```
+
+Now, find the IP address for react-form, open it on your browser and publish messages to each topic (A, B and C).
+
+## Confirm subscribers received the messages.
+
+Switch back to `namespace-a`:
+```
+kubectl config set-context --current --namespace=namespace-a
+```
+
+Find the POD names:
+```
+kubectl get pod # Copy POD names and use in the next commands.
+```
+
+Display logs:
+```
+kubectl logs node-subscriber-XYZ node-subscriber
+kubectl logs python-subscriber-XYZ python-subscriber
+```
+

```

Comment:
If everything went well, what should users see here?

Reply:
Described output similar to how the sample does.